ssl/statem/extensions_clnt.c

```diff
@@ -652,6 +741,12 @@ EXT_RETURN tls_construct_ctos_key_share(SSL *s, WPACKET *pkt,
     const uint16_t *pgroups = NULL;
     uint16_t curve_id = 0;
 
+    /* Send an empty key share if we are allowing a non-DHE key exchange method, and are not retrying the ClientHello. */
+    if (s->s3->group_id == 0 && s->psksession != NULL && s->options & SSL_OP_ALLOW_NO_DHE_KEX) {
+        /* Don't add a key share */
```

Comment:
The comment above says send an empty key share, but here we say don't add a key share at all. These are different things.